src/LuaColor.cpp

```diff
@@ -0,0 +1,213 @@
+// Copyright © 2008-2019 Pioneer Developers. See AUTHORS.txt for details
+// Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+#include "LuaColor.h"
+#include "LuaUtils.h"
+#include "libs.h"
+
+inline void sanity_check(float r, float g, float b, float a)
+{
+	r = std::max(std::min(r, float(255.0)), float(0.0));
+	g = std::max(std::min(g, float(255.0)), float(0.0));
+	b = std::max(std::min(b, float(255.0)), float(0.0));
+	a = std::max(std::min(a, float(255.0)), float(0.0));
+}
+
+static int l_color_new(lua_State *L)
+{
+	LUA_DEBUG_START(L);
+	double r = luaL_checknumber(L, 1);
+	double g = luaL_checknumber(L, 2);
+	double b = luaL_checknumber(L, 3);
+	double a;
+	if (lua_isnumber(L, 4)) a = luaL_checknumber(L, 4);
+	else a = 255.0;
+	sanity_check(r, g, b, a);
+	LuaColor::PushToLua(L, Color(r, g, b, a));
+	LUA_DEBUG_END(L, 1);
+	return 1;
+}
+
+static int l_color_call(lua_State *L)
+{
+	LUA_DEBUG_START(L);
+	double r = luaL_checknumber(L, 2);
+	double g = luaL_checknumber(L, 3);
+	double b = luaL_checknumber(L, 4);
+	double a;
+	if (lua_isnumber(L, 5)) a = luaL_checknumber(L, 5);
+	else a = 255.0;
+	sanity_check(r, g, b, a);
+	Color c = Color(r, g, b, a);
```

Comment:
You appear to be doing a no-op here, and I'd strike it from the function entirely.